pkg/controller/core.oam.dev/v1alpha2/application/suite_test.go

```diff
@@ -55,6 +57,7 @@ import (
 // These tests use Ginkgo (BDD-style Go testing framework). Refer to
 // http://onsi.github.io/ginkgo/ to learn more about Ginkgo.
 var cfg *rest.Config
+var recorder = record.NewFakeRecorder(10000)
```

Comment:
I noticed the fakeRecorder is very simple, it doesn't include the appname in the event message, so you can hardly distinguish them . I suggest you can re-write that package here. At least include the object name.

```

func (f *FakeRecorder) Event(object runtime.Object, eventtype, reason, message string) {
	if f.Events != nil {
		f.Events <- fmt.Sprintf("%s %s %s %s", object.Name(),eventtype, reason, message)
	}
}

```